File: Mini-Projects/JAVA/Selection_sort.java
package searching_and_sorting;

import java.util.Scanner;

public class Selection_sort {
	
	static Scanner s = new Scanner(System.in);

	//function to take input
		public static int[] takeInput(){
			
			int size = s.nextInt();
			int[] input = new int[size];
			for(int i = 0; i < size; i++){
				input[i] = s.nextInt();
			}
			return input;
		}
		//end
		
		//function to sort
		public static void selectionSort(int [] arr)
		{
			for(int i = 0; i < arr.length - 1; i++)
			{
				int min = arr[i];
				int minIndex = i;
				
				for(int j = i + 1; j < arr.length; j++)
				{
					if(arr[j] < min)
					{
						min = arr[j];
						minIndex = j;
					}
				}
					if(minIndex != i)
					{
						arr[minIndex] = arr[i];
						arr[i] = min;
					}
			}
		}
		//end
		
		//function to print array
		public static void printArray(int[] arr) {	
			for (int i = 0; i < arr.length; i++) {
				System.out.print(arr[i] + " ");
			}

			System.out.println();
		}
		//end
		
		
	public static void main(String[] args) {
		
		int[] arr = takeInput();
		selectionSort(arr);
		printArray(arr);
	}

}

File: Mini-Projects/CPP/selectionsort.cpp
#include<bits/stdc++.h>
using namespace std;

int A[10];

void swap(int *x, int *y)
{
	int temp = *x;
	*x = *y;
	*y = temp;
}

void selectionSort(int arr[10])
{
	int min;

	for(int i=0; i<10; i++)
	{
		min = i;

		for(int j=i+1; j<10; j++)
		{
			if(arr[j] < arr[min])
			{
				min = j;
			}
		}

		swap(&arr[min], &arr[i]);
	}
}

void print(int arr[10])
{
	cout<<"The sorted array is:"<<endl;

	for(int i=0; i<10; i++)
	{
		cout<<arr[i]<<" ";
	}
}

void input(int arr[10])
{
	for(int i=0; i<10; i++)
	{
		cin>>arr[i];
	}
}

int main()
{
	input(A);
	selectionSort(A);
	print(A);
	return 0;
}